evalml/model_understanding/prediction_explanations/_user_interface.py

```diff
@@ -741,7 +741,9 @@ def make_dict(self, index, y_pred, y_true, scores, dataframe_index):
 
         return {
             "probabilities": pred_values,
-            "predicted_value": _make_json_serializable(self.predicted_values[index]),
+            "predicted_value": _make_json_serializable(
```

Comment:
Calling iloc will get the positional index. Since `.predict()` now preserves the original index, which may be non-numeric, we need to call `iloc` here!